### PR TITLE
S4   see applicants attending an assessment day

### PR DESCRIPTION
**Pull request by oritchie77**
added applicant modal to the event page, added copy all button to event details.

### CODE REVIEW

File: public/js/eventsPage.js [hunk 1]
```diff
@@ -299,64 +305,80 @@ async function eventGenerator(event, hiringPartners, applicants) {
...
-async function getHiringPartners() {
-
-    let response = await fetch('./api/getHiringPartnerInfo', {
-        credentials: 'same-origin',
-        headers: {
-            'Accept': 'application/json',
-            'Content-Type': 'application/json',
-        },
-        method: 'get',
+function addEventListenerForCopyEmailsButton(event, applicants) {
```
Comment (norayna) [inline public/js/eventsPage.js:308]: rename to `addEventListenersToCopyEmailsButtons`

File: public/js/eventsPage.js [hunk 2]
```diff
@@ -299,64 +305,80 @@ async function eventGenerator(event, hiringPartners, applicants) {
...
+    let assessmentApplicants
+    let copyEmailsButtons = document.querySelectorAll('.copy-emails-button')
+    copyEmailsButtons.forEach((button) => {
+        button.addEventListener('click', (event) => {
+            event.stopPropagation()
```
Comment (norayna) [inline public/js/eventsPage.js:313]: probably don't need this line of code

File: public/js/eventsPage.js [hunk 3]
```diff
@@ -299,64 +305,80 @@ async function eventGenerator(event, hiringPartners, applicants) {
...
+            assessmentApplicants = applicants.filter(applicant => applicant.assessmentDay === event.target.dataset.id)
+            let attendeeEmails = ''
+            assessmentApplicants.forEach((applicant) => {
+                attendeeEmails += applicant.email + '; '
+            })
+            navigator.clipboard.writeText(attendeeEmails)
+            })
     })
-    return await response.json()
 }
```
Comment (norayna) [inline public/js/eventsPage.js:322]: why is this deleted?

File: public/js/applicantModal.js [hunk 4]
```diff
@@ -50,7 +49,8 @@ function copyToClipboard(element) {
     document.querySelector("button.clipboard").innerText = 'Copied';
 }
 
-$(document).ready(function () {
+export function displayApplicantModal() {
```
Comment (norayna) [inline public/js/applicantModal.js:52]: rename to `addEventListenersToDisplayApplicantModal`

Comment (oritchie77) [reply public/js/applicantModal.js:52]: done

File: public/js/applicantModal.js [hunk 5]
```diff
@@ -50,7 +49,8 @@ function copyToClipboard(element) {
...
+    $(document).ready(function () {
     $(".myBtn").click(function () {
         let url = './api/getApplicant/' + this.dataset.id
```
[the remark this answers is not in the document]
Comment (oritchie77) [reply public/js/applicantModal.js:55]: done

File: templates/applicants.phtml [hunk 6]
```diff
@@ -7,7 +7,7 @@
     ?>
     <link rel="stylesheet" type="text/css" href="./css/style.css">
     <script src="./js/populateDropdowns.js" defer></script>
-    <script defer src="./js/applicantModal.js"></script>
+    <script defer src="./js/applicantModal.js" type="module"></script>
```
Comment (norayna) [inline templates/applicants.phtml:10]: does this have to be `type="module"` on this page?

File: public/scss/_events.scss [hunk 7]
```diff
@@ -64,6 +74,16 @@
...
 
+.event-attendees-table {
+    background-color: #ffffff;
+}
+
+.email-column-header {
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+}
```
Comment (norayna) [inline public/scss/_events.scss:85]: delete all that because it causes a weird double border, instead put `float: right` on the button itself

File: public/scss/_genericStyles.scss [hunk 8]
```diff
@@ -69,4 +69,10 @@ input[type="submit"] {
 
 table {
     width: 100%;
+}
+
+.myBtn {
+    &:hover {
+      cursor: pointer;
+    }
```
Comment (norayna) [inline public/scss/_genericStyles.scss:77]: instead of this, the `<a>` should have a `href`

File: templates/events.phtml [hunk 9]
```diff
@@ -6,21 +6,25 @@
     require 'partials/layout/head.phtml';
     echo \Portal\ViewHelpers\LayoutViewHelper::outputTitle('Events');
     ?>
-    <script defer src='./js/eventsPage.js'></script>
+    <script defer src='./js/eventsPage.js' type="module"></script>
+    <script src="./js/populateDropdowns.js" defer></script>
+
```
Comment (norayna) [inline templates/events.phtml:11]: remove?

**Review by norayna — CHANGES_REQUESTED** [review]
asking for changesss

**Review by norayna — APPROVED** [review]
